### ISSUE TITLE
Add capability to observe the cookie state

### ISSUE THREAD

**Issue by miherlosev** [DevExpress/testcafe#4428]
TestCafe uses the own cookie processing mechanism. It leads to a user cannot see the full cookie list same as in the browser development tools.
![image](https://user-images.githubusercontent.com/4133518/67667501-04e83380-f97f-11e9-9a06-010836acdd68.png)

We need to provide such information. It will be useful for complex authentication cases.

https://github.com/DevExpress/testcafe/issues/1721#issuecomment-514227960

**Comment by alexey-lin** [comment]
https://github.com/DevExpress/testcafe/issues/4425

**Comment by AndreyBelym** [comment]
https://stackoverflow.com/questions/59957576/how-to-get-httponly-secure-cookies-testcafe/59984223#59984223

**Comment by austinhinderer** [comment]
I'm still hitting this issue at the end of 2020. Unable to read httpOnly cookies in testCafe.

**Comment by Farfurix** [comment]
@austinhinderer 

Hello,

Unfortunately, I cannot give you any time estimates on when this enhancement will be addressed.

**Comment by deepaneela** [comment]
It will be helpful to have this feature which will get all the cookies including httpOnly, secure  basically all cookies we see in dev tools.

**Comment by jamesyoung1337** [comment]
For the moment, you can use a work-around to access httpOnly cookies:

[My answer on Stack Overflow](https://stackoverflow.com/questions/59957576/how-to-get-httponly-secure-cookies-testcafe/)

Here it is again anyway:

```
import { Selector, RequestLogger } from 'testcafe';

// Want everything from example.com
const logger = RequestLogger(/example.com/, {
   logResponseHeaders: true,
   logResponseBody: true
   // Do not set stringify response body if your response
   // comes as gzip or brotli or whatever, instead you
   // will need to use Node's zlib to unzip it and read it
});

fixture `My test`
   .page `https://example.com`
   .requestHooks(logger);

test(`Log in and retrieve cookies`, async t => {
   
   await t
   // Do things here
   .click(...)

   const requests = logger.requests
   for (const req of requests) {
       console.log('Headers: ', req.response.headers)

       for (const cookie of req.response.headers['set-cookie']) {
          // You will have to parse the cookie yourself
          console.log(cookie)
       }
   }
});
```

You may need to wrap your code in a promise and await on it, especially if you access the body as JSON like I do in addition to the cookies, otherwise code after this request logger block may be executed before you read the body.

**Comment by helen-dikareva** [comment]
An advanced cookie management API was released in v1.19.0. We added new flexible cross-browser methods that allow you to manipulate browser cookies with ease even if they have the `HttpOnly` flag specified.

The following example demonstrates how to obtain all cookies from all testing pages:

```js
const allCookies =  await t.getCookies();
```

Learn more about each cookie method in our docs:
[getCookies](https://testcafe.io/documentation/403873/reference/test-api/testcontroller/getcookies)
[setCookies](https://testcafe.io/documentation/403872/reference/test-api/testcontroller/setcookies)
[deleteCookies](https://testcafe.io/documentation/403874/reference/test-api/testcontroller/deletecookies)
